src/export/StructureDefinitionExporter.ts

```diff
@@ -482,6 +482,20 @@ export class StructureDefinitionExporter implements Fishable {
           // Note: newElement() method automatically adds the new element to its structDef.elements
           const newElement = structDef.newElement(rule.path);
           newElement.applyAddElementRule(rule, this);
+          const lastIdPart = newElement.id.slice(newElement.id.lastIndexOf('.') + 1);
+          // see http://hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition eld-19 and eld-20
+          const nameErrors: string[] = [];
+          if (!/^[^\s\\.,:;\\'"\/|?!@#$%&*()\[\]{}]+(\[x\])?$/.test(lastIdPart)) {
```

Comment:
Ah, it's even more than that: I don't need any backslashes before the `.`, since that's already considered as a literal when used in a character class.

Reply:
Oh even better! Good catch.